src/ast.rs

```diff
@@ -417,6 +430,13 @@ impl Statement {
                     "The drop keyword can only be applied to function calls.".to_string(),
                 ))
             }
+            Statement::OptionalBlock(_) => {
+                return Err(ParseErrorMsg::new(
```

Comment:
I would add an error test for this.

Reply:
So, interestingly, it appears that this error is actually unreachable.  The parser permits drop before a let binding or function call, but at a parser level, drop cannot appear before an if block or an optional block.

It would be fairly straightforward to turn these parse errors into compile errors:

```
<start: @L> "drop" <end: @R> <mut s: StmtBody> ";"
```

Could just become:

```
<start: @L> "drop" <end: @R> <mut s:Stmt>
```

In addition to allowing drop in front of optional and if blocks, this would also allow the perverse "drop drop drop drop some_statement();" at a parser level, which seems mildly undesirable to me (I think it would be functionally identical to just one drop).

The advantage of allowing it at the parser, is that these error messages are more helpful than the "Unexpected word optional" that the parser returns here.  So it's probably worth letting this all pass through the parser and catching it here.

Thoughts?

Reply:
My first reaction is if we let it past the parser there may be weird corner cases we need to deal with later.  Like I didn't even think about stuff like "drop drop drop".  If you think it won't get out of hand, I am good with it going past the parser.

Reply:
Okay, I looked at this some more, and decided that really it makes more sense for drop to apply only to calls at a parser level, which simplifies the code in the ast for handling all of it, removing these error conditions.  I pushed that change as a new commit here, and I think it cleans all of this up and addresses this concern.